validation/link-validator.js

```diff
@@ -0,0 +1,172 @@
+      linkLog[`relative-source-text-${index}`] = link?.children[0].value;
+    }
+    try {
+      updateUrl = new URL(updateLink.url);
+    } catch (e) {
+      updateUrl = false;
+      linkLog[`relative-update-link-${index}`] = link.url;
+      linkLog[`relative-update-text-${index}`] = link?.children[0].value;
+    }
+
+    // Partial matches are not fully qualified urls
+    if (!sourceUrl || !updateUrl) {
+      linkLog[`partialUrlMatch-${index}`] = link.url === updateLink.url;
+    }
+
+    linkLog[`sourceLink-${index}`] = link.url;
+    linkLog[`updatedLink-${index}`] = link.children[0].value;
+    linkLog[`sourceText-${index}`] = updateLinks[index].url;
+    linkLog[`updatedText-${index}`] = updateLinks[index]?.children[0]?.value;
+    linkLog[`linksMatch-${index}`] = link.url === updateLink.url;
+    linkLog[`hashMatch-${index}`] = sourceUrl ? sourceUrl.hash === updateUrl.hash : '';
+    linkLog[`hostMatch-${index}`] = sourceUrl ? sourceUrl.host === updateUrl.host : '';
+    linkLog[`pathMatch-${index}`] = sourceUrl ? sourceUrl.pathname === updateUrl.pathname : '';
+    linkLog[`searchMatch-${index}`] = sourceUrl ? sourceUrl.search === updateUrl.search : '';
+    linkLog[`textMatch-${index}`] = link?.children[0]?.value === updateLink?.children[0]?.value;
+  });
+
+  return ['Deep Compare Links', LINKS_DO_NOT_MATCH, path, { log: linkLog }];
+}
+
+/**
+ * Does an initial check for link matching, if it finds issue, it runs the deepCompare function
+ *
+ * @param {list of source links} sourceLinks
+ * @param {list of updated links} updatedLinks
+ * @param {path to files} path
+ * @returns Returns a message for reporter
+ */
+export function compareLinkLists(sourceLinks, updatedLinks, path) {
+  console.log(`Comparing source and update files at this path: ${path}`);
+  // If not the same length, something is wrong from the start
+  if (sourceLinks.length !== updatedLinks.length) {
+    return ['Compare Links', 'list length', LENGTHS_DO_NOT_MATCH];
+  }
+
+  const linksMatch = !sourceLinks.map((link, i) => {
+    const updated = updatedLinks[i];
+    return link.url === updated.url
+    && link?.children[0]?.value === updated?.children[0]?.value;
+  }).includes(false);
+
+  if (!linksMatch) {
+    return deepCompare(sourceLinks, updatedLinks, path);
+  }
+
+  return ['Compare Links', LINKS_MATCH, path];
+}
+
+/**
+ * Runs the primary migration and returns a message for reporter.
+ * Additional logic for code failing conditions
+ *
+ * @param {list of paths} list
+ * @param {path to md files} mdPath
+ * @param {ExcelReporter} reporter
+ */
+export async function validateMigratedPageLinks(list, mdPath, reporter) {
+  const listData = await loadListData(list);
+
+  for (const path of listData) {
+    const pathToSourceMd = path.endsWith('/') ? `${mdPath}/source${path}index.md` : `${mdPath}/source${path}.md`;
+    const pathToUpdateMd = path.endsWith('/') ? `${mdPath}/updated${path}index.md` : `${mdPath}/updated${path}.md`;
```

Comment:
The "source" and "updated" values should be a parameter or a const.

Reply:
I think I got your meaning, please check again when you get a sec.